Bling joystick integration

binnur opened the pull request:
- Fixed Robot.java LED state calls
- Refactored SetBlingStateCommand as LEDCommands
- Updated LED subsystem and periodic()
    * Moved the dashboard update to setBlingState() to reduce unnecessary network table
    updates
    * Added error states to dashboard's 'LED state'
- Renamed bling states
    * renamed BLING_COMMAND_STARTUP to BLING_COMMAND_TELEOP
    * repurposed BLING_COMMAND_LOADING for BLING_COMMAND_DRIVE_SLOW
    * BLING_COMMAND_DEFAULT for BLING_COMMAND_EJECT
    * BLING_STATE_PICKUP to BLING_STATE_INTAKE
    * BLING_COMMAND_SHOOTING to BLING_COMMAND_LAUNCH
- Additional comment, etc. cleanups

binnur committed:
Updated BLING_STATEs and joystick integration

- Fixed Robot.java LED state calls
- Refactored SetBlingStateCommand as LEDCommands
- Updated LED subsystem and periodic()
    * Moved the dashboard update to setBlingState() to reduce unnecessary network table
    updates
    * Added error states to dashboard's 'LED state'
- Renamed bling states
    * renamed BLING_COMMAND_STARTUP to BLING_COMMAND_TELEOP
    * repurposed BLING_COMMAND_LOADING for BLING_COMMAND_DRIVE_SLOW
    * BLING_COMMAND_DEFAULT for BLING_COMMAND_EJECT
    * BLING_STATE_PICKUP to BLING_STATE_INTAKE
    * BLING_COMMAND_SHOOTING to BLING_COMMAND_LAUNCH
- Additional comment, etc. cleanups

binnur commented:
@j-james @pietroglyph as bling integration impacts RobotContainer.java, I recommend merging sooner than later. With that said, I'll do my best to maintain parity w/ joystick changes.

Notes:
- Verify LED::SetBlingState() operation correctly sends the bling state change
- Need to validate bling transitions, especially based on triggers -- see TODOs in RobotContainer
- I will review for other inline bling integrations we may need.

binnur commented:
@randomgrace I made several changes to the LED system, fyi.

randomgrace reviewed (approved):
Nice - I like the commands integration.

binnur committed:
Turning off LEDs during control panel actions

When control panel mechanism is raised, LEDs are OFF.
When control panel mechanism is lowered, TELEOP animation is run.
